ddl/partition.go

```diff
@@ -1335,6 +1335,8 @@ func checkExchangePartitionRecordValidation(w *worker, pt *model.TableInfo, inde
 		} else if len(pi.Columns) == 1 {
 			sql = buildCheckSQLForRangeColumnsPartition(pi, index, schemaName, tableName)
 		}
+	case model.PartitionTypeList:
+		sql = fmt.Sprintf("select 1 from `%s`.`%s` where %s = %s limit 1", schemaName.L, tableName.L, pi.Columns[index], pi.Definitions[index].InValues[index][0])
```

Comment:
Shouldn't the SQL be `select 1 from %s.%s where %s not in (ALL OF THE IN VALUES FROM TARGET PARTITION) limit 1`?